galette/lib/Galette/Entity/Adherent.php

```diff
@@ -857,9 +858,60 @@ public function getDues(): string
                 $ret = _T("No longer member");
             }
         }
+
+        if (self::getCountDonations() > 0) {
+            if ($never_contributed) {
+                $ret = _T("Never contributed, but a donation");
+            } else {
+                $ret .= ' - '._T("Donator");
+            }
+        }
         return $ret;
     }
 
+    /**
+     * Get donation count in the current year
+     *
+     * @return int count 0..x
+     */
+    public function getCountDonations()
```

Comment:
Types must be declared
```suggestion
    public function getCountDonations(): int
```